prime/src/components/borrow/ManageAccountTransactionButton.tsx

```diff
@@ -197,11 +196,48 @@ export function ManageAccountTransactionButton(props: ManageAccountTransactionBu
 
   if (writeAsset0Allowance.isError) writeAsset0Allowance.reset();
   if (writeAsset1Allowance.isError) writeAsset1Allowance.reset();
-  if (contract.isError || contract.isSuccess) setTimeout(contract.reset, 500);
 
   // check whether we're prepared to send a transaction (independent of whether transaction will succeed/fail)
   const canConstructTransaction = actionOutputs.findIndex((o) => o.actionArgs === undefined) === -1;
 
+  const actionIds = actionOutputs.map((o) => getFrontendManagerCodeFor(o.actionId));
+  const actionArgs = actionOutputs.map((o) => o.actionArgs!);
+  let positions = '0';
+
+  // if Uniswap positions are being changed, make sure to send an updated array of positions
+  if (actionIds.includes(4) || actionIds.includes(5)) {
+    positions = zip(accountState.uniswapPositions);
+  }
+
+  // provide ante if necessary
+  const shouldProvideAnte =
+    accountEtherBalance &&
+    accountEtherBalance.value.toNumber() < ANTE &&
+    (accountState.liabilities.amount0 > 0 || accountState.liabilities.amount1 > 0);
+
+  const calldata = canConstructTransaction
+    ? ethers.utils.defaultAbiCoder.encode(['uint8[]', 'bytes[]', 'uint144'], [actionIds, actionArgs, positions])
+    : null;
+
+  const { config: contractConfig } = usePrepareContractWrite({
+    address: accountAddress,
+    abi: MarginAccountAbi,
+    functionName: 'modify',
+    chainId: activeChain.id,
+    args: [ALOE_II_FRONTEND_MANAGER_ADDRESS, calldata, [UINT256_MAX, UINT256_MAX]],
```

Comment:
I believe the args at the end should be bools, not `UINT256_MAX`es. That should reduce gas slightly. Another thing we can do to reduce gas is to conditionally set them to true/false. The first arg only needs to be `true` when our list of actions contains:
- Removing token0 collateral
- Swapping token0 for token1

Similarly, the second arg only needs to be `true` when:
- Removing token1 collateral
- Swapping token1 for token0

Doing this will avoid unnecessary `approve`s in the contracts.